src/api/downloadFile.js

```diff
@@ -7,8 +7,10 @@ import { getAuthHeader, getFullUrl } from '../utils/url';
 import userAgent from '../utils/userAgent';
 
 export default (src: string, auth: Auth) => {
+  const fullUrlWithAPIKey = `${getFullUrl(src, auth.realm)}?api_key=${auth.apiKey}`;
```

Comment:
Is it guaranteed that the URL will not already have a `?` in it?

Reply:
In `js.js` we do have this code:

```js
const delimiter = img.src.includes('?') ? '&' : '?';
img.src += `${delimiter}api_key=${auth.apiKey}`;
```

Reply:
Yeah, that logic seems more robust to me.

Too bad it isn't currently easy to share code between `js.js` and the rest of the app -- but fortunately this is only a line or so and unlikely to change, so it's not a big problem to duplicate it. :slightly_smiling_face: